talktorials/11_online_apis/11a_obtaining_structures.ipynb

```diff
@@ -0,0 +1,2939 @@
+    "        family = random.choice(families)\n",
+    "        kinase = random.choice(_kinases_from_family(family))\n",
+    "        try:\n",
+    "            molcomplex, protein, ligands = _protein_and_ligand_structure(kinase.kinase_ID)\n",
+    "        except:\n",
+    "            attempts -= 1\n",
+    "            time.sleep(1)\n",
+    "        else:                   \n",
+    "            print(\"Chosen\", kinase.name, \"kinase with ID\", kinase.kinase_ID, \"from family\", family)\n",
+    "            return molcomplex, protein, ligands\n",
+    "    print(\"Could not find a valid kinase. Try again!\")\n",
+    "    return None, None, None\n",
+    "\n",
+    "\n",
+    "def step_01(family):\n",
+    "    \"\"\"\n",
+    "    Given a kinase family name (check `_all_kinase_families()`), retrieve a random structure.\n",
+    "    \"\"\"\n",
+    "    attempts = 20\n",
+    "    while attempts:  # some kinase IDs cannot be found in the structure list...\n",
+    "        kinase = random.choice(_kinases_from_family(family))\n",
+    "        try:\n",
+    "            molcomplex, protein, ligands = _protein_and_ligand_structure(kinase.kinase_ID)\n",
+    "        except:\n",
+    "            attempts -= 1\n",
+    "            time.sleep(1)\n",
+    "        else:                   \n",
+    "            print(\"Chosen\", kinase.name, \"kinase with ID\", kinase.kinase_ID, \"from family\", family)\n",
+    "            return molcomplex, protein, ligands\n",
+    "    print(\"Could not find a valid kinase. Try again!\")\n",
+    "    return None, None, None"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's check if this works. Let's say you want a random kinase from a random family, you can use `step_01_random()` like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Chosen KLIFS entry with PDB ID 5u6b with chain C and alternate model \n",
+      "Chosen AXL kinase with ID 398 from family Axl\n"
+     ]
+    }
+   ],
+   "source": [
+    "molcomplex, protein, ligands = step_01_random()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['Clc1c(F)ccc2c1[C@H](Oc3nc(cnc3N)C=4C(=NN(C4CN(C2=O)C)CCO)C)C']"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "ligands"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's preview the protein with `nglview` and the ligands with `rdkit`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nglview as nv\n",
```

Comment:
I think 11a needs the "!pip install nglview" cell (11c has it but users mostly will start with 11a).